src/wallet/scriptpubkeyman.cpp

```diff
@@ -732,12 +787,27 @@ bool LegacyScriptPubKeyMan::AddWatchOnly(const CScript& dest, int64_t nCreateTim
 void LegacyScriptPubKeyMan::SetHDChain(const CHDChain& chain, bool memonly)
 {
     LOCK(cs_wallet);
-    if (!memonly && !WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain))
-        throw std::runtime_error(std::string(__func__) + ": writing chain failed");
+    if (!memonly) {
+        if (!WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain)) {
+            throw std::runtime_error(std::string(__func__) + ": writing chain failed");
+        }
+        if (!hdChain.seed_id.IsNull()) {
+            AddInactiveHDChain(hdChain, false);
```

Comment:
sanity check: should this be `hdChain` or `chain`?

Reply:
It is supposed to be `hdChain`.

`SetHDChain` happens when we change `hdChain`. We want to store what is currently `hdChain` as an inactive seed before we switch it out for the new one. So this is correct, we add it as an inactive chain before it is set to the new `chain`.